airbyte-config/models/src/main/java/io/airbyte/config/AirbyteProtocolConverters.java

```diff
@@ -77,9 +81,31 @@ public static Schema toSchema(AirbyteCatalog catalog) {
           .withName(airbyteStream.getName())
           .withFields(list.stream().map(item -> new Field()
               .withName(item.getKey())
-              .withDataType(DataType.valueOf(item.getValue().get("type").asText().toUpperCase()))
+              .withDataType(singerTypesToDataType(item.getValue().get("type")))
               .withSelected(true)).collect(Collectors.toList()));
     }).collect(Collectors.toList()));
   }
 
+  /**
+   * Singer tends to have 2 types for fields one of which is null. The null is pretty irrelevant, so
+   * look at types and find the first non-null one and use that.
+   *
+   * @param singerTypes - list of types discovered by singer.
+   * @return reduce down to one type which best matches the field's data type
+   */
+  private static DataType singerTypesToDataType(JsonNode node) {
```

Comment:
This is not singer right this is jsonschema

Reply:
yes. copy paste error. i stole this function from our other converter, but yes you are right. this is jsonschema type to data type.